website/docs/d/upgrade_prepare_ready.html.markdown

```diff
@@ -0,0 +1,25 @@
+---
+subcategory: "Beta"
+layout: "nsxt"
+page_title: "NSXT: upgrade_prepare_ready"
+description: A data source to check if NSX Manager is ready for upgrade.
+---
+
+# nsxt_upgrade_prepare_ready
+
+This data source provides information about the upgrade readiness of NSX Manager.
+The resource creation will succeed only if there are no errors in precheck and
+all warnings have been acknowledged.
+
+## Example Usage
+
+```hcl
+data "nsxt_upgrade_prepare_ready" "test" {
+  upgrade_prepare_id = nsxt_upgrade_prepare.test.id
```

Comment:
I think we should explain that this data source might depend on  `nsxt_upgrade_precheck_acknowledge` if those exist. I would add this to the example.

Reply:
done